goosebit/ui/bff/software/routes.py

```diff
@@ -24,13 +24,23 @@
     "",
     dependencies=[Security(validate_user_permissions, scopes=["software.read"])],
 )
-async def software_get(dt_query: Annotated[DataTableRequest, Depends(parse_datatables_query)]) -> BFFSoftwareResponse:
-    def search_filter(search_value: str):
-        return Q(uri__icontains=search_value) | Q(version__icontains=search_value)
+async def software_get(
+    request: Request, dt_query: Annotated[DataTableRequest, Depends(parse_datatables_query)]
+) -> BFFSoftwareResponse:
+    filters: list[Q] = []
+
+    def search_filter(search_value):
+        base_filter = Q(Q(uri__icontains=search_value), Q(version__icontains=search_value), join_type="OR")
+        return Q(base_filter, *filters, join_type="AND")
 
     query = Software.all().prefetch_related("compatibility")
 
-    return await BFFSoftwareResponse.convert(dt_query, query, search_filter)
+    uuids = request.query_params.get("uuids")
+    if uuids:
+        hardware = await Hardware.filter(devices__uuid__in=uuids.split(",")).distinct()
+        filters.append(Q(*[Q(compatibility__id=c.id) for c in hardware], join_type="AND"))
```

Comment:
A bit more optimized for the DB
```
        filters.append(Q(compatibility__id__in=list(map(lambda hw:hw.id, hardware_ids))))
```

Reply:
I think this is an OR join, not an AND join.  My goal here was to ensure the compatibility was valid for all selected devices, I think this change would show all software compatible with any selected device?

Reply:
You are correct.